HybridWebView (.NET 10): initialization customization and platform view access

davidortinau opened the pull request:
This PR adds .NET 10 guidance to customize initialization and access platform web views in `HybridWebView`.

Summary of changes:
- Adds a new ">= net-maui-10.0" section "Customize initialization and access platform web views" to `docs/user-interface/controls/hybridwebview.md`.
- Documents Windows `MauiHybridWebView.RunAfterInitialize` for safe access to `CoreWebView2` after initialization.
- Shows how to access and tweak the platform views on Android (`android.webkit.WebView`) and iOS/Mac Catalyst (`WKWebView`) after creation via `HandlerChanged`.
- Includes an advanced custom handler example to provide a `WKWebViewConfiguration` at creation time when platform options must be set before the view is created.
- Updates `ms.date`.

Motivation / verification:
- HybridWebView does not expose app-facing initializing/initialized events like BlazorWebView. The recommended approach is to use platform view access and, on Windows, `RunAfterInitialize`.
- Verified against MAUI net10 source (handlers for Windows/Android/iOS/macOS) and `MauiHybridWebView` helper.

Related upstream references:
- dotnet/maui#28876 (intercept all web requests – context for recent HybridWebView docs)
- dotnet/maui#29790 (HybridWebView TypeScript)
- dotnet/maui#29794 (expose hybrid script)

Docs hygiene:
- Moniker blocks are balanced.
- Examples compile logically; no xref changes needed.

Once merged, I'll update the tracker entry to link to this PR and mark status accordingly.

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/user-interface/controls/hybridwebview.md](https://github.com/dotnet/docs-maui/blob/a9b7749aff98ad1d4f97636a1704326950256116/docs/user-interface/controls/hybridwebview.md) | [docs/user-interface/controls/hybridwebview](https://review.learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/hybridwebview?branch=pr-en-us-3003) |


<!-- PREVIEW-TABLE-END -->

davidortinau committed:
HybridWebView (.NET 10): document platform initialization customization and platform view access; add Windows RunAfterInitialize guidance; custom handler example for WKWebViewConfiguration; update ms.date

BethMassi commented:
For the .NET 10 version of this article, we also need to remove the requirement that the hybridwebview.js file needs to be manually created. All you need to reference now in your web content is `<script src="_framework/hybridwebview.js"></script>` as this is now embedded. 

I'm not sure how to structure the monikers in this doc to do this properly though.

jfversluis commented on docs/user-interface/controls/hybridwebview.md:
```diff
@@ -681,6 +681,100 @@ The `window.HybridWebView.InvokeDotNet` JavaScript function invokes a specified
+        };
+
+        // Recreate the platform view with your configuration
+        var webview = new MauiHybridWebView(this, CoreGraphics.CGRect.Empty, config);
```
This block is only shown for the .NET 10 moniker so assumption is that only people using .NET 10 will see this

jfversluis commented on docs/user-interface/controls/hybridwebview.md:
```diff
@@ -681,6 +681,100 @@ The `window.HybridWebView.InvokeDotNet` JavaScript function invokes a specified
+
+While <xref:Microsoft.Maui.Controls.HybridWebView> doesn’t expose app-facing initializing/initialized events like <xref:Microsoft.AspNetCore.Components.WebView.Maui.BlazorWebView>, you can still customize the underlying platform web views and run code after they’re ready:
+
+- Windows (WebView2): the platform view is <xref:Microsoft.Maui.Platform.MauiHybridWebView>, which inherits `WebView2` and adds `RunAfterInitialize(Action)` so you can safely access `CoreWebView2` once it’s ready.
```
```suggestion
- Windows (WebView2): the platform view is <xref:Microsoft.Maui.Controls.HybridWebView>, which inherits `WebView2` and adds `RunAfterInitialize(Action)` so you can safely access `CoreWebView2` once it’s ready.
```

jfversluis committed:
Apply suggestion from @jfversluis

BethMassi commented:
@jfversluis perhaps something like this would work?

 A simple app might have the following files and contents:
    - *Resources\Raw\wwwroot\index.html* with content for the main UI:
```  
        <!DOCTYPE html>
        <html lang="en" xmlns="http://www.w3.org/1999/xhtml">
        <head>
            <meta charset="utf-8" />
            <title></title>
            <link rel="icon" href="data:,">
            <link rel="stylesheet" href="styles/app.css">
```
::: moniker range="<=net-maui-9.0"
`<script src="scripts/HybridWebView.js"></script>`
::: moniker-end
::: moniker range=">=net-maui-10.0"
`<script src="_framework/hybridwebview.js"></script>`
::: moniker-end

rest of the index.html goes here....

Then for this section we just exclude it completely for .NET 10+:

::: moniker range="<=net-maui-9.0"

- *Resources\Raw\wwwroot\scripts\HybridWebView.js* with the standard <xref:Microsoft.Maui.Controls.HybridWebView> JavaScript library:
```
        window.HybridWebView = {
            "Init": function Init() {
                function DispatchHybridWebViewMessage(message) {
                    const event = new CustomEvent("HybridWebViewMessageReceived", { detail: { message: message } });
                    window.dispatchEvent(event);
                }
```
rest of HybridWebView.js ....

::: moniker-end

jfversluis committed:
Update HybridWebView.md with versioning info

Added moniker range for .NET MAUI versioning in HybridWebView documentation.

jfversluis committed:
Merge branch 'main' into pr11-hybridwebview-init-customization-net10-from-main

jfversluis committed:
Update hybridwebview.md

jfversluis committed:
Fix moniker syntax in hybridwebview.md

jfversluis committed:
Update HybridWebView script references for .NET MAUI

jfversluis committed:
Enhance HybridWebView documentation with new JS features

Updated HTML content for HybridWebView with new JavaScript functions for message logging and invoking C# methods. Adjusted script references for different .NET MAUI versions.

jfversluis committed:
Fix moniker-end formatting in hybridwebview.md
